### PR TITLE
Processing data models

### PR DESCRIPTION
**Pull request by gandradep**
- Create models
- Use models to insert data
Command Prompt - psql  -U postgres 1_26_2023 3_48_37 PM
- Create custom methods

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
```
Comment (mirouhml) [inline app/models/comment.rb]: [OPTIONAL]

- [ ] You can add `after_save` here to call the `update_post_comment_counter`method every time a new comment has been saved to database, then make the `update_post_comment_counter` method private.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_post_likes_counter
+    post.update(likesCounter: post.likes.count)
+  end
```
Comment (mirouhml) [inline app/models/like.rb]: [OPTIONAL]

- [ ] You can add `after_save` here to call the `update_post_likes_counter` method every time a new like has been saved to database, then make the `update_post_likes_counter` method private.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_post_counter
+    author.update(postsCounter: author.posts.count)
+  end
```
Comment (mirouhml) [inline app/models/post.rb]: [OPTIONAL]

- [ ] You can add `after_save` here to call the `update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

**Review by mirouhml — APPROVED** [review]
# APPROVED 🟢

Hi @gandradep, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gandradep** [comment]
@mirouhml Thanks for the review and the suggestions. I will add them right now before I merge it.

**Pull request merged**